src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultFallbackCapabilitiesFilterResolver.cs

```diff
@@ -0,0 +1,247 @@
+                case MSLSPMethods.WorkspacePullDiagnosticName:
+                    return CheckPullDiagnosticCapabilities;
+
+                default:
+                    return FallbackCheckCapabilties;
+            }
+        }
+
+        private static bool CheckImplementationCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.ImplementationProvider != null;
+        }
+
+        private bool CheckTypeDefinitionCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.TypeDefinitionProvider != null;
+        }
+
+        private static bool CheckFindAllReferencesCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.ReferencesProvider != null;
+        }
+
+        private static bool CheckRenameCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.RenameProvider != null;
+        }
+
+        private static bool CheckSignatureHelpCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.SignatureHelpProvider != null;
+        }
+
+        private static bool CheckWillSaveCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.TextDocumentSync?.WillSave == true;
+        }
+
+        private static bool CheckWillSaveWaitUntilCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.TextDocumentSync?.WillSaveWaitUntil == true;
+        }
+
+        private static bool CheckRangeFormattingCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.DocumentRangeFormattingProvider != null;
+        }
+
+        private static bool CheckWorkspaceSymbolCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.WorkspaceSymbolProvider != null;
+        }
+
+        private static bool CheckOnTypeFormattingCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.DocumentOnTypeFormattingProvider != null;
+        }
+
+        private static bool CheckFormattingCapabilities(JToken token)
+        {
+            var serverCapabilities = token.ToObject<ServerCapabilities>();
+
+            return serverCapabilities?.DocumentFormattingProvider != null;
```

Comment:
>DocumentFormattingProvider [](start = 39, length = 26)

Out of curiosity, how did you determine these? In our delegating layer, we're basically just copying over from vslanguageserverclient, eg:

        protected override bool CheckCapability(ServerCapabilities serverCapabilities)
        {
            return serverCapabilities.DocumentFormattingProvider?.Match(
                            boolValue => boolValue,
                            documentFormattingOptions => documentFormattingOptions != null) ?? false;
        }

Reply:
> Out of curiosity, how did you determine these?

tried coming up with them against the spec; however, looks like I definitely fumbled the multi-type scenarios where it can be either or. I'll fix